Framework/NexusGeometry/src/NexusGeometrySave.cpp

```diff
@@ -778,47 +784,50 @@ class NexusGeometrySaveImpl {
     return childGroup;
   }
 
-  /* For NXinstrument parent (component info root). Produces an NXmonitor
-   * groups from Component info, and saves it in the parent
+  /* For NXinstrument parent. Produces an NXmonitor
+   * group from Component info, and saves it in the parent
    * group, along with the Nexus compliant datasets, and metadata stored in
    * attributes to the new group.
    *
    * Saves detector-spectra mappings too
    *
-   * @param parentGroup : parent group in which to write the NXinstrument
+   * @param parentGroup : NXinstrument parent group in which to write the NXmonitor
    * group.
    * @param compInfo : componentInfo object.
    * @param monitorId :  ID of the specific monitor.
    * @param index :  index of the specific monitor in the Instrument cache.
    * @param mappings : Spectra to detector mappings
    */
   void monitor(const H5::Group &parentGroup, const Geometry::ComponentInfo &compInfo, const int monitorId,
-               const size_t index, const SpectraMappings &mappings) {
+               const size_t index, const std::optional<SpectraMappings> &mappings) {
 
     auto childGroup = monitor(parentGroup, compInfo, monitorId, index);
-    // Additional mapping information written.
-    writeDetectorCount(childGroup, mappings);
-    // Note that the detector list is the same as detector_number, but it is
-    // ordered by spectrum index 0 - N, whereas detector_number is just written
-    // out in the order the detectors are encountered in the bank.
-    writeDetectorList(childGroup, mappings);
-    writeDetectorIndex(childGroup, mappings);
-    writeSpectra(childGroup, mappings);
+
+    if (mappings) {
+      // Additional mapping information written.
+      writeDetectorCount(childGroup, *mappings);
+      // Note that the detector list is the same as detector_number, but it is
+      // ordered by spectrum index 0 - N, whereas detector_number is just written
+      // out in the order the detectors are encountered in the bank.
+      writeDetectorList(childGroup, *mappings);
+      writeDetectorIndex(childGroup, *mappings);
+      writeSpectra(childGroup, *mappings);
+    }
```

Comment:
I may have missed it, but why is mapping optional now?

Reply:
There is an (external) _signature_ where there's no mapping info, and this is the _core_ code section (internal).  That would be used when the NeXus file contains _only_ the instrument.  (For example, via the `SaveNexusGeometry` algorithm.)